app/managers/performance_metrics_manager/index.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {AppState, type AppStateStatus} from 'react-native';
+import performance from 'react-native-performance';
+
+import Batcher from './performance_metrics_batcher';
+
+type Target = 'HOME' | 'CHANNEL' | 'THREAD' | undefined;
+type MetricName = 'mobile_channel_switch' |
+    'mobile_team_switch';
+
+class PerformanceMetricsManager {
+    private target: Target;
+    private batchers: {[serverUrl: string]: Batcher} = {};
+    private hasRegisteredLoad = false;
+    private lastAppStateIsActive = AppState.currentState === 'active';
+
+    constructor() {
+        AppState.addEventListener('change', (appState) => this.onAppStateChange(appState));
+    }
+
+    private onAppStateChange(appState: AppStateStatus) {
+        const isAppStateActive = appState === 'active';
+        if (this.lastAppStateIsActive !== isAppStateActive && !isAppStateActive) {
+            for (const batcher of Object.values(this.batchers)) {
+                batcher.forceSend();
+            }
+        }
+        this.lastAppStateIsActive = isAppStateActive;
+    }
+
+    private ensureBatcher(serverUrl: string) {
+        if (this.batchers[serverUrl]) {
+            return this.batchers[serverUrl];
+        }
+
+        this.batchers[serverUrl] = new Batcher(serverUrl);
+        return this.batchers[serverUrl];
+    }
+
+    public setLoadTarget(target: Target) {
+        this.target = target;
+    }
+
+    public finishLoad(location: Target, serverUrl: string) {
+        if (this.target !== location || this.hasRegisteredLoad) {
+            return;
+        }
+
+        const measure = performance.measure('mobile_load', 'nativeLaunchStart');
+        this.ensureBatcher(serverUrl).addToBatch({
+            metric: 'mobile_load',
+            value: measure.duration,
+            timestamp: Date.now(),
```

Comment:
Now that we have that library, could thse use `performance.timeOrigin + measure.startTime`?

Reply:
Agree

Reply:
I got many issues to make this sum up. This is the values I get from the library (on a real android device):
```
{"dateNow": 1718095572718, "dur": 401.2168229999952, "origin": 553634.5101, "startTime": 623363.507261, "sum": 1176998.017361, "timestamp": 1718095642446.9973}
```
The `timestamp` one is calculated as stated here: https://github.com/oblador/react-native-performance/blob/master/packages/react-native-performance/README.md

So... the only value that makes sense to use here is `Date.now`. The timestamp is also not so important, since it is only used for bucketing, and we are not going so fine grain down to seconds.

If you have any other idea, I am happy to try :smile:

Reply:
Huh, I feel like that doesn't match how the browser API works, but if that's how they say to do it, that's fine by me